2.0RC5:   java.lang.IllegalStateException - com.owncloud.android.ui.preview.PreviewImageFragment.onActivityCreated

```
java.lang.RuntimeException: 
  at android.app.ActivityThread.performLaunchActivity (ActivityThread.java:2762)
  at android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:2848)
  at android.app.ActivityThread.-wrap12 (ActivityThread.java)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1552)
  at android.os.Handler.dispatchMessage (Handler.java:102)
  at android.os.Looper.loop (Looper.java:154)
  at android.app.ActivityThread.main (ActivityThread.java:6334)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:886)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:776)
Caused by: java.lang.IllegalStateException: 
  at com.owncloud.android.ui.preview.PreviewImageFragment.onActivityCreated (PreviewImageFragment.java:225)
  at android.support.v4.app.Fragment.performActivityCreated (Fragment.java:2257)
  at android.support.v4.app.FragmentManagerImpl.moveToState (FragmentManager.java:1367)
  at android.support.v4.app.FragmentManagerImpl.moveFragmentToExpectedState (FragmentManager.java:1569)
  at android.support.v4.app.FragmentManagerImpl.moveToState (FragmentManager.java:1636)
  at android.support.v4.app.FragmentManagerImpl.dispatchActivityCreated (FragmentManager.java:2977)
  at android.support.v4.app.FragmentController.dispatchActivityCreated (FragmentController.java:192)
  at android.support.v4.app.FragmentActivity.onStart (FragmentActivity.java:603)
  at android.support.v7.app.AppCompatActivity.onStart (AppCompatActivity.java:178)
  at com.owncloud.android.ui.activity.BaseActivity.onStart (BaseActivity.java:189)
  at com.owncloud.android.ui.activity.DrawerActivity.onStart (DrawerActivity.java:1273)
  at com.owncloud.android.ui.activity.FileActivity.onStart (FileActivity.java:188)
  at com.owncloud.android.ui.preview.PreviewImageActivity.onStart (PreviewImageActivity.java:173)
  at android.app.Instrumentation.callActivityOnStart (Instrumentation.java:1322)
  at android.app.Activity.performStart (Activity.java:6760)
  at android.app.ActivityThread.performLaunchActivity (ActivityThread.java:2725)
```

--- a/src/main/java/com/owncloud/android/ui/preview/PreviewImageFragment.java
+++ b/src/main/java/com/owncloud/android/ui/preview/PreviewImageFragment.java
@@ -34,6 +34,7 @@ import android.os.Build;
 import android.os.Bundle;
 import android.os.Process;
 import android.support.annotation.DrawableRes;
+import android.support.annotation.NonNull;
 import android.support.annotation.StringRes;
 import android.support.v4.app.FragmentStatePagerAdapter;
 import android.util.DisplayMetrics;
@@ -121,7 +122,7 @@ public class PreviewImageFragment extends FileFragment {
      *                              {@link FragmentStatePagerAdapter}
      *                              ; TODO better solution
      */
-    public static PreviewImageFragment newInstance(OCFile imageFile, boolean ignoreFirstSavedState) {
+    public static PreviewImageFragment newInstance(@NonNull OCFile imageFile, boolean ignoreFirstSavedState) {
         PreviewImageFragment frag = new PreviewImageFragment();
         Bundle args = new Bundle();
         args.putParcelable(ARG_FILE, imageFile);
@@ -152,7 +153,7 @@ public class PreviewImageFragment extends FileFragment {
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         Bundle args = getArguments();
-        setFile((OCFile) args.getParcelable(ARG_FILE));
+        setFile(args.getParcelable(ARG_FILE));
         // TODO better in super, but needs to check ALL the class extending FileFragment;
         // not right now
 
@@ -218,14 +219,7 @@ public class PreviewImageFragment extends FileFragment {
                 mIgnoreFirstSavedState = false;
             }
         }
-        if (getFile() == null) {
-            throw new IllegalStateException("Instanced with a NULL OCFile");
-        }
-        if (!getFile().isDown()) {
-            throw new IllegalStateException("There is no local file to preview");
-        }
     }
-
 
     /**
      * {@inheritDoc}
@@ -236,18 +230,16 @@ public class PreviewImageFragment extends FileFragment {
         outState.putParcelable(PreviewImageFragment.EXTRA_FILE, getFile());
     }
 
-
     @Override
     public void onStart() {
         super.onStart();
-        if (getFile() != null) {
+        if (getFile() != null && !getFile().isDown()) {
+            showErrorMessage(R.string.preview_image_error_no_local_file);
+        } else if (getFile() != null) {
             mLoadBitmapTask = new LoadBitmapTask(mImageView);
-            //mLoadBitmapTask.execute(new String[]{getFile().getStoragePath()});
-//            mLoadBitmapTask.execute(getFile().getStoragePath());
             mLoadBitmapTask.execute(getFile());
         }
     }
-
 
     @Override
     public void onStop() {
@@ -557,7 +549,7 @@ public class PreviewImageFragment extends FileFragment {
             if (result.bitmap != null || result.drawable != null) {
                 showLoadedImage(result);
             } else {
-                showErrorMessage();
+                showErrorMessage(mErrorMessageId);
             }
             if (result.bitmap != null && mBitmap != result.bitmap) {
                 // unused bitmap, release it! (just in case)
@@ -632,11 +624,11 @@ public class PreviewImageFragment extends FileFragment {
             mImageView.setVisibility(View.VISIBLE);
 
         }
+    }
 
-        private void showErrorMessage() {
-            mImageView.setBackgroundColor(Color.TRANSPARENT);
-            setMessageForMultiList(mErrorMessageId, R.string.preview_sorry, R.drawable.file_image);
-        }
+    private void showErrorMessage(@StringRes int errorMessageId) {
+        mImageView.setBackgroundColor(Color.TRANSPARENT);
+        setMessageForMultiList(R.string.preview_sorry, errorMessageId, R.drawable.file_image);
     }
 
     private void setMultiListLoadingMessage() {
@@ -655,9 +647,9 @@ public class PreviewImageFragment extends FileFragment {
             mMultiListMessage.setText(message);
             mMultiListIcon.setImageResource(icon);
 
+            mMultiListMessage.setVisibility(View.VISIBLE);
             mMultiListIcon.setVisibility(View.VISIBLE);
             mMultiListProgress.setVisibility(View.GONE);
-
         }
     }
 

--- a/src/main/java/com/owncloud/android/ui/preview/PreviewImageFragment.java
+++ b/src/main/java/com/owncloud/android/ui/preview/PreviewImageFragment.java
@@ -233,9 +233,9 @@ public class PreviewImageFragment extends FileFragment {
     @Override
     public void onStart() {
         super.onStart();
-        if (getFile() != null && !getFile().isDown()) {
+        if (getFile() == null || !getFile().isDown()) {
             showErrorMessage(R.string.preview_image_error_no_local_file);
-        } else if (getFile() != null) {
+        } else {
             mLoadBitmapTask = new LoadBitmapTask(mImageView);
             mLoadBitmapTask.execute(getFile());
         }

--- a/src/main/java/com/owncloud/android/ui/preview/PreviewImagePagerAdapter.java
+++ b/src/main/java/com/owncloud/android/ui/preview/PreviewImagePagerAdapter.java
@@ -138,22 +138,19 @@ public class PreviewImagePagerAdapter extends FragmentStatePagerAdapter {
     
     public Fragment getItem(int i) {
         OCFile file = mImageFiles.get(i);
-        Fragment fragment = null;
+        Fragment fragment;
         if (file.isDown()) {
-            fragment = PreviewImageFragment.newInstance(file,
-                    mObsoletePositions.contains(Integer.valueOf(i)));
+            fragment = PreviewImageFragment.newInstance(file, mObsoletePositions.contains(i));
             
-        } else if (mDownloadErrors.contains(Integer.valueOf(i))) {
+        } else if (mDownloadErrors.contains(i)) {
             fragment = FileDownloadFragment.newInstance(file, mAccount, true);
             ((FileDownloadFragment)fragment).setError(true);
-            mDownloadErrors.remove(Integer.valueOf(i));
+            mDownloadErrors.remove(i);
             
         } else {
-            fragment = FileDownloadFragment.newInstance(
-                    file, mAccount, mObsoletePositions.contains(Integer.valueOf(i))
-            );
+            fragment = FileDownloadFragment.newInstance(file, mAccount, mObsoletePositions.contains(i));
         }
-        mObsoletePositions.remove(Integer.valueOf(i));
+        mObsoletePositions.remove(i);
         return fragment;
     }
 

--- a/src/main/res/values/strings.xml
+++ b/src/main/res/values/strings.xml
@@ -342,9 +342,10 @@
     <string name="conflict_use_local_version">local version</string>
     <string name="conflict_use_server_version">server version</string>
 
-    <string name="preview_sorry">Sorry.</string>
+    <string name="preview_sorry">Sorry</string>
     <string name="preview_image_description">Image preview</string>
     <string name="preview_image_error_unknown_format">Unable to show image</string>
+    <string name="preview_image_error_no_local_file">There is no local file to preview</string>
 
     <string name="error__upload__local_file_not_copied">%1$s could not be copied to %2$s local folder</string>
     <string name="prefs_instant_upload_path_title">Instant upload folder</string>

